### PR TITLE
LPS-195154 Adding tests to create an action

### PR DESCRIPTION
**Pull request by JerryAdrianoo**
**Description:**
Adding tests to create an Action

**Jira Ticket:**
https://liferay.atlassian.net/browse/LPS-195154

### CODE REVIEW

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
cc: @CarlosBrichete  these failures are not related 😄

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 1]
```diff
@@ -418,4 +534,184 @@ definition {
...
+		task ("And clicks on plus button") {
+			LexiconEntry.gotoAdd();
+		}
+
+		task ("Then the URL field is mandatory") {
+			AssertElementPresent(
+				key_fieldName = "URL",
+				locator1 = "DataSet#MANDATORY_FIELD");
+		}
+	}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:701]: Just checking the mandatory field is not enough, we should try to save and check that the action cannot be saved without filling the field

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 2]
```diff
@@ -418,4 +534,184 @@ definition {
...
+		task ("And clicks on Create Item Action button") {
+			LexiconEntry.gotoAdd();
+		}
+
+		task ("Then the Message Type field is not mandatory") {
+			AssertElementNotPresent(
+				key_fieldName = "Message Type",
+				locator1 = "DataSet#MANDATORY_FIELD");
+		}
+	}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:715]: Just checking the mandatory field is not enough, we should try to save and check that the action can be saved without filling the field

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 3]
```diff
@@ -418,4 +534,184 @@ definition {
...
+				key_fieldLabel = "Type",
+				locator1 = "Select#GENERIC_SELECT_FIELD");
+		}
+
+		task ("Then a dropdown is displayed containing the options: Link") {
+			AssertElementPresent(
+				key_typeName = "Link",
+				locator1 = "Select#TYPE_SELECTED");
+		}
+	}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:687]: The Type filed cannot be empty by definition, so this test is not needed. We can just remove it

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 4]
```diff
@@ -418,4 +534,184 @@ definition {
...
+				locator1 = "DataSet#MANDATORY_FIELD");
+		}
+
+		task ("And Label field is localized ") {
+			AssertElementPresent(
+				key_field = "Label",
+				key_localizedLanguage = "en_US",
+				locator1 = "DataSet#LOCALIZED_FIELD");
+		}
+	}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:655]: Just checking the mandatory field is not enough, we should try to save and check that the action cannot be saved without filling the field

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 5]
```diff
@@ -418,4 +534,184 @@ definition {
...
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "Link");
+		}
+
+		task ("Then the Confirmation Message field is present") {
+			AssertElementPresent(
+				key_placeholder = "Add a message here.",
+				locator1 = "FrontendDataSet#PLACEHOLDER");
+		}
+	}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:628]: Just checking the mandatory field is not enough, we should try to save and check that the action can be saved without filling the field

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 6]
```diff
@@ -418,4 +534,184 @@ definition {
...
+			Select(
+				key_fieldLabel = "Icon",
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "accessibility");
+
+			AssertElementPresent(
+				key_text = "pt-br",
+				locator1 = "Icon#ANY");
+		}
+	}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:607]: Just checking the mandatory field is not enough, we should try to save and check that the action cannot be saved without filling the field. Also, please check it manually but I think that now the icon is mandatory and cannot be empty. Fix the test accordingly

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 7]
```diff
@@ -418,4 +534,184 @@ definition {
...
+				locator1 = "Icon#ANY_N");
+		}
+
+		task ("Then the message “This key allows the action to be performed when the permission is associated with it” is displayed.") {
+			AssertElementPresent(
+				key_title = "",
+				locator1 = "DataSet#FILTER_TOOLTIP",
+				value1 = "This key is used to display the action if the user has the permission for it by checking if the key is present in the data item's actions array.");
+		}
+	}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:570]: Just checking the mandatory field is not enough, we should try to save and check that the action can be saved without filling the field

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]
Please check my requests

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
cc @CarlosBrichete finally was finished. Can I forward it?

**Review by CarlosBrichete — APPROVED** [review]
LGTM :)

**Comment by JerryAdrianoo** [comment]
ci:forward

**Pull request closed without merging**